src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -360,6 +400,81 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
                     return attributeList.WithoutTrivia();
                 }
             }));
+        }
+
+        private static async Task RefactorInitializersAsync(
+            INamedTypeSymbol type,
+            SolutionEditor solutionEditor,
+            ImmutableArray<IPropertySymbol> positionalParameters,
+            CancellationToken cancellationToken)
+        {
+            var symbolReferences = await SymbolFinder
+                .FindReferencesAsync(type, solutionEditor.OriginalSolution, cancellationToken).ConfigureAwait(false);
+            var referenceLocations = symbolReferences.SelectMany(reference => reference.Locations);
+            var projLookup = referenceLocations.ToLookup(refLoc => refLoc.Document.Project.Id);
+            foreach (var (projID, projLocs) in projLookup)
+            {
+                // organize by project first, so we can solve one project at a time
+                var project = solutionEditor.OriginalSolution.GetRequiredProject(projID);
+                if (project.Language != LanguageNames.CSharp)
+                {
+                    // skip non-CSharp projects that reference records
+                    continue;
+                }
+
+                var compilation = await project
+                    .GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var docLookup = projLocs.ToLookup(refLoc => refLoc.Document.Id);
+                foreach (var (docID, docLocs) in docLookup)
+                {
+                    var documentEditor = await solutionEditor
+                        .GetDocumentEditorAsync(docID, cancellationToken).ConfigureAwait(false);
+                    var syntaxTree = await documentEditor.OriginalDocument
+                        .GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var semanticModel = compilation.GetSemanticModel(syntaxTree);
+                    var nodes = docLocs
+                        // we should find the identifier node of an object creation expression
+                        .Select(refLoc => refLoc.Location.FindNode(cancellationToken).Parent)
+                        .OfType<ObjectCreationExpressionSyntax>();
+                    foreach (var node in nodes)
+                    {
+                        var (constructorArgs, nodesToRemove) =
+                            ConvertToRecordHelpers.GetConstructorArgumentsFromObjectCreation(
+                                node, positionalParameters, semanticModel, cancellationToken);
```

Comment:
we have ConvertTORecordHelpers *and* ConvertToRecordCommon?

Reply:
Yes. ConvertToRecordHelpers does most of the operation work, such as determining whether hashcode, equals, and constructors are ok to delete, and gets some information sometimes from their operations. ConvertToRecordCommon is the code that actually performs the refactoring (now also offered as a code fix)

Reply:
They both are in the same folder and namespace. I could definitely merge the files as they are both static classes; the only reason I haven't is to distinguish the (slightly) different kinds of work they do.

Reply:
I suspect calling "ConvertToRecordCommon" instead something like "ConvertToRecordEngine" or something clarifies it's the core engine or something. But yes, if they're both static methods maybe just merge...but do that as a separate PR.